extension/partiql-extension-ion/src/boxed_ion.rs

```diff
@@ -143,44 +187,70 @@ impl BoxedVariant for BoxedIon {
             BoxedIonValue::Sequence(_) => {
                 DatumCategoryOwned::Sequence(DatumSeqOwned::Dynamic(self))
             }
-            BoxedIonValue::Value(elt) => match elt.ion_type() {
-                IonType::List => DatumCategoryOwned::Sequence(DatumSeqOwned::Dynamic(self)),
-                IonType::SExp => DatumCategoryOwned::Sequence(DatumSeqOwned::Dynamic(self)),
-                IonType::Null => DatumCategoryOwned::Null,
-                IonType::Struct => DatumCategoryOwned::Tuple(DatumTupleOwned::Dynamic(self)),
-                _ => DatumCategoryOwned::Scalar(DatumValueOwned::Value(self.into_value())),
-            },
+            BoxedIonValue::Value(elt) => {
+                if elt.is_null() {
+                    DatumCategoryOwned::Null
+                } else {
+                    match elt.ion_type() {
+                        IonType::List => DatumCategoryOwned::Sequence(DatumSeqOwned::Dynamic(self)),
+                        IonType::SExp => DatumCategoryOwned::Sequence(DatumSeqOwned::Dynamic(self)),
+                        IonType::Null => DatumCategoryOwned::Null,
+                        IonType::Struct => {
+                            DatumCategoryOwned::Tuple(DatumTupleOwned::Dynamic(self))
+                        }
+                        _ => DatumCategoryOwned::Scalar(DatumValueOwned::Value(self.into_value())),
+                    }
+                }
+            }
         }
     }
 }
 
+impl PartialEq<Self> for BoxedIon {
+    fn eq(&self, other: &Self) -> bool {
+        self.doc.eq(&other.doc)
+    }
+}
+
+impl Eq for BoxedIon {}
+
+impl PartialOrd for BoxedIon {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+impl Ord for BoxedIon {
+    fn cmp(&self, other: &Self) -> Ordering {
+        // TODO lowering just to compare is costly... Either find a better way, or lift this out of the extension
+        self.lower().unwrap().cmp(&other.lower().unwrap())
```

Comment:
Probably we should address? (unless it's addressed in a subsequent PR)

Reply:
It is indeed replaced in #543 : https://github.com/partiql/partiql-lang-rust/pull/543/files#diff-b95d0de35899342f30d2531c8cd3728a958d846d163f0c154c204f7aeb8979ceL222-L225